crates/ruma-state-res/src/event_auth.rs

```diff
@@ -417,24 +415,17 @@ fn valid_membership_change<E: Event>(
             )?)
         })?;
 
-    let power_levels = power_levels_event.map_or_else(
-        || Ok::<_, Error>(PowerLevelsEventContent::default()),
-        |power_levels| {
-            serde_json::from_value::<PowerLevelsEventContent>(power_levels.content())
-                .map_err(Into::into)
-        },
-    )?;
+    let power_levels: PowerLevelsEventContent = match power_levels_event {
+        Some(ev) => serde_json::from_value(ev.content())?,
+        None => PowerLevelsEventContent::default(),
+    };
```

Comment:
Jeez, `map_or_else` is terrible, it's like twice as much code yuck. Good call on all these